### PR TITLE
Update pixel.wp.com to HTTPS

### PR DESCRIPTION
**Pull request by bhubbard**
Sets all pixel.wp.com urls/resources to always be https.

### CODE REVIEW

**Comment by jeherve** [comment]
Before merging, we'll need to make sure performance remains the same when loading that tracking pixel over https.

@jonathansadowski Does that sound like a good change for you?

**Comment by jonathansadowski** [comment]
There shouldn't be any side-effects to this change. The parameters of the pixel are parsed equally regardless of protocol. 👍

**Comment by jeherve** [comment]
@jonathansadowski I was more thinking about the speed when serving over HTTP vs. HTTPS, on domains that weren't optimized by Systems to serve content over HTTPS.

**Comment by jonathansadowski** [comment]
Sorry, @jeherve, thanks for clarifying. I did some basic benchmarking, and you're correct, there's a not-insignificant performance cost to this change (the request taking in the ballpark of 30% longer in my testing)

There are some cases where we already (potentially) absorb this cost. For example, `do_server_side_stat`, uses `build_stats_url`, whose protocol choice depends on `is_ssl()`.

File: class.jetpack.php [hunk 1]
```diff
@@ -3757,7 +3757,7 @@ static function build_stats_url( $args ) {
 		 */
 		$base_url = apply_filters(
 			'jetpack_stats_base_url',
-			set_url_scheme( 'http://pixel.wp.com/g.gif' )
+			set_url_scheme( 'https://pixel.wp.com/g.gif' )
```
Comment (jeherve) [inline class.jetpack.php]: If we want to use `HTTPS` all the time, we'll need to get rid of `set_url_scheme()` here.

**Comment by jeherve** [comment]
I'm looking into this, and will report back once I have some news.

**Comment by jeherve** [comment]
Turns out everything will be fine with HTTPS, it shouldn't have any significant performance impact.

**Pull request merged**